### PR TITLE
[Feature] Audit Purging

### PR DESCRIPTION
**Pull request by Orrison**
Add automated Audit purging based on admin configurable retention period

### CODE REVIEW

File: app-modules/audit/database/migrations/2023_08_16_025823_create_audit_settings.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 return new class () extends SettingsMigration {
     public function up(): void
     {
-        $this->migrator->add('audit.retention_duration', 129600);
+        $this->migrator->add('audit.retention_duration', 90);
```
Comment (dgoetzit) [inline app-modules/audit/database/migrations/2023_08_16_025823_create_audit_settings.php:9]: I'm wondering if we also need to define a unit, or make this setting more explicit in the code that it's in days.

File: app-modules/engagement/src/Models/EngagementFile.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ public function students(): MorphToMany
             table: 'engagement_file_entities',
             foreignPivotKey: 'engagement_file_id',
             relatedPivotKey: 'entity_id',
-            relation: 'student',
+            relation: 'engagementFiles',
```
Comment (dgoetzit) [inline app-modules/engagement/src/Models/EngagementFile.php:76]: I think this should remain `student`

**Review by dgoetzit — CHANGES_REQUESTED** [review]
This looks good - just have a couple of suggestions.

File: app-modules/engagement/src/Models/EngagementFile.php (hunk 2 above)
Comment (Orrison) [reply app-modules/engagement/src/Models/EngagementFile.php:76]: It seems to break when I set it as `student`

File: app-modules/audit/database/migrations/2023_08_16_025823_create_audit_settings.php (hunk 1 above)
Comment (Orrison) [reply app-modules/audit/database/migrations/2023_08_16_025823_create_audit_settings.php:9]: Hmm, like perhaps a comment within the settings class for this?

File: app-modules/engagement/src/Models/EngagementFile.php (hunk 2 above)
Comment (dgoetzit) [reply app-modules/engagement/src/Models/EngagementFile.php:76]: That's odd 🤔

Comment (Orrison) [reply app-modules/engagement/src/Models/EngagementFile.php:76]: I'll continue to look into it more. But this change has it working now

File: app-modules/audit/database/migrations/2023_08_16_025823_create_audit_settings.php (hunk 1 above)
Comment (Orrison) [reply app-modules/audit/database/migrations/2023_08_16_025823_create_audit_settings.php]: Fixed, went with option #1 for now

**Pull request merged**